lib/fork.js

```diff
@@ -18,13 +18,29 @@ module.exports = function (args) {
 
 	var ps = childProcess.fork(babel, args, options);
 
+	var killed = false;
+	var killedPromise = new Promise(function (resolve) {
+		ps.on('exit', function (code) {
+			killed = true;
+			resolve(code);
+		});
+	});
+
+	function kill() {
+		if (!killed) {
+			ps.kill();
+		}
+		return killedPromise;
+	}
+
 	var promise = new Promise(function (resolve, reject) {
 		ps.on('results', function (results) {
+			results.kill = kill;
```

Comment:
We don't have a reference to the promise in [`cli.js#exit()`](https://github.com/jamestalmage/ava/blob/7087a08dda47b5d7549d4b93f90fe22d49816acf/cli.js#L136), only the result. Hence the need to attach it to the result.

> That way, we won't have weird stuff in test results.

Not sure how this would create "weird stuff" in test results.